server/data.py

```diff
@@ -268,50 +256,57 @@ def _count_gt_for_write(self, variants, data, filters, chips, data_type):
         id_index = list(filtered_basic_info.index) 
         for i, d in enumerate(data):
             # get indices of het/hom individuals in genotype data
-            # het_i, hom_alt_i, info = self._get_het_hom_index(d, id_index, filters['gtgp'] == 'gt', filters['gpThres'], len(data) == 1)
             if data_type == 'imputed':
-                het_i, hom_alt_i, info = self._get_het_hom_index(d, id_index, filters['gtgp'] == 'gt', filters['gpThres'], len(data) == 1)
+                het_i, hom_alt_i, wt_hom_i, missing_i, info = self._get_het_hom_index(d, id_index, filters['gtgp'] == 'gt', filters['gpThres'], len(data) == 1)
             else:
-                het_i, hom_alt_i, info = self._get_het_hom_index(d, id_index, True, None, False)
+                het_i, hom_alt_i, wt_hom_i, missing_i, info = self._get_het_hom_index(d, id_index, True, None, False)
             gt = [element.split(':')[0] for element in d ]
             gt_arr = np.array(gt)
             gt_het = list(gt_arr[het_i])
             gt_hom = list(gt_arr[hom_alt_i])
+            gt_wt_hom = list(gt_arr[wt_hom_i])
+            gt_missing = list(gt_arr[missing_i])
 
             # subset dataframe for het/hom individuals, copy needed as this will be mutated
             het = info_orig.iloc[het_i].copy()
             hom_alt = info_orig.iloc[hom_alt_i].copy()
+            wt_hom = info_orig.iloc[wt_hom_i].copy()
+            missing = info_orig.iloc[missing_i].copy()
 
             # extract gt probs and gts
             if data_type == 'imputed':
                 gt_probs = [element.split(':')[2] for element in d ]
                 gt_probs_arr = np.array(gt_probs)
                 gt_probs_het = list(gt_probs_arr[het_i])
                 gt_probs_hom = list(gt_probs_arr[hom_alt_i])
+                gt_probs_wt_hom = list(gt_probs_arr[wt_hom_i])
                 het['three_gt_probs'] = gt_probs_het
                 hom_alt['three_gt_probs'] = gt_probs_hom
+                wt_hom['three_gt_probs'] = gt_probs_wt_hom                
```

Comment:
again too big change for now but all these elements appearing on if conditions instead of proper encapsulation to classes make this very unmaintainable and error prone. 
I know you inherited this in the state but I think we should take time on between releases to now do some significant refactoring.